emcc.py

```diff
@@ -1624,10 +1624,7 @@ def is_supported_link_flag(f):
 
       # WASM_ASYNC_COMPILATION and SWAPPABLE_ASM_MODULE do not have a meaning in MINIMAL_RUNTIME (always async)
       if not shared.Settings.MINIMAL_RUNTIME:
-        if shared.Settings.WASM_ASYNC_COMPILATION == 1:
-          # async compilation requires a swappable module - we swap it in when it's ready
-          shared.Settings.SWAPPABLE_ASM_MODULE = 1
-        else:
+        if shared.Settings.WASM_ASYNC_COMPILATION and shared.Settings.WASM2JS:
           # if not wasm-only, we can't do async compilation as the build can run in other
           # modes than wasm (like asm.js) which may not support an async step
           shared.Settings.WASM_ASYNC_COMPILATION = 0
```

Comment:
It seems like this line was alwasy previously a no-op.  I guess this code was the result of some kind of merge conflict in the past because it makes no sense to be as is.

Reply:
This change basically moves the check for WASM_ASYNC_COMPILATION below this block that disables is.